test/sqllogictest/list.slt

```diff
@@ -1243,81 +1319,347 @@ SELECT length(LIST['1','2']::text)
 query error Cannot call function length\(string list\): arguments cannot be implicitly cast to any implementation's parameters; try providing explicit casts
 SELECT length(LIST['1','2'])
 
-# 🔬🔬 list_prepend
+# 🔬🔬 text to list
 
-# 🔬🔬🔬 element + list
+# 🔬🔬🔬 text to int list
```

Comment:
Would you want to add tests like the following to this section and the `text to text list` section?
```
query error ...
SELECT * FROM unnest('{1,2,3}'::int list list)
```

Reply:
Ah, maybe I should have a more general test section; this wasn't meant to test anything specifically about `int list`; just to show that these were in fact `list`s  that could be unnested.